packages/ckeditor5-utils/src/dom/position.ts

```diff
@@ -99,12 +100,44 @@ export function getOptimalPosition( { element, target, positions, limiter, fitIn
 
 	let bestPosition: Position;
 
+	// Get intersection of all scrollable ancestors of `target`.
+	const allScrollableAncestors = getScrollableAncestors( target as HTMLElement );
+	const ancestorsIntersection: Rect | null = allScrollableAncestors
+		.reverse()
+		.map( item => new Rect( item as HTMLElement ) )
+		.reduce( ( ancestorIntersectionAccumulator, ancestorRect ): Rect => {
+			if ( !ancestorIntersectionAccumulator || ancestorIntersectionAccumulator.top === undefined ) {
```

Comment:
When would `top` be `undefined`? Which use-case is it?

Reply:
Is this to cover cases when `getIntersection()` returns `null` that you ignore by casting `as Rect` in the last line?

Reply:
for example:

```
new Rect( global.document )
------
Rect {top: undefined, right: undefined, bottom: undefined, left: undefined, width: undefined, height: undefined}
```